### PR TITLE
Listing support for multi-tenant classes

### PR DESCRIPTION
**Pull request by aliszka**
### What's being changed:
Adds support for listing objects (with or without class specified) of MT classes.

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [x] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

**Review by antas-marcin — APPROVED** [review]

File: adapters/repos/db/search.go [hunk 1]
```diff
@@ -267,23 +267,41 @@ func (db *DB) objectSearch(ctx context.Context, offset, limit int,
...
+		for _, index := range db.indices {
+			// TODO support all additional props
+			res, _, err := index.objectSearch(ctx, totalLimit,
+				filters, nil, sort, nil, additional, nil, tenant, 0)
+			if err != nil {
+				// TODO find better way to recognise particular errors
+				if errors.As(err, &objects.ErrInvalidUserInput{}) {
+					// validation failed (either MT class without tenantKey or non-MT class with tenantKey)
+					if strings.Contains(err.Error(), "has multi-tenancy enabled, but request was without tenant") ||
+						strings.Contains(err.Error(), "has multi-tenancy disabled, but request was with tenant") {
```
Comment (dirkkul) [inline adapters/repos/db/search.go:284]: make constants out of those errors?

File: adapters/repos/db/index.go [hunk 2]
```diff
@@ -320,41 +320,36 @@ func indexID(class schema.ClassName) string {
...
-	if err != nil {
-		return "", errors.Wrap(err, "parse id as uuid")
-	}
-
-	uuidBytes, _ := uuid.MarshalBinary() // cannot error
-
-	return ss.PhysicalShard(uuidBytes), nil
-}
-
 func (i *Index) determineObjectShard(id strfmt.UUID, tenantKey string, ss *sharding.State) (string, error) {
```
Comment (dirkkul) [inline adapters/repos/db/index.go:323]: can we break this function up in a "getShardingState" function and then call `shardFromTenantKey` and `shardFromUUID` directly? I noticed that some callers of this function already have a `if tenant != ""` around it and that makes the flow somewhat confusing

File: adapters/repos/db/search.go (hunk 1 above)
Comment (dirkkul) [reply adapters/repos/db/search.go:284]: There is also `error.Is` which might make the code easier (but error handling in go suuuuuucks)

File: adapters/repos/db/search.go [hunk 3]
```diff
@@ -267,23 +267,41 @@ func (db *DB) objectSearch(ctx context.Context, offset, limit int,
...
+						continue
+					}
+					// tenantKey not supported by class
+					if strings.Contains(err.Error(), "no tenant found with key") {
```
Comment (dirkkul) [inline adapters/repos/db/search.go:288]: same

**Review by dirkkul — APPROVED** [review]

File: adapters/repos/db/search.go (hunk 1 above)
Comment (aliszka) [reply adapters/repos/db/search.go:284]: I saved it for later. Definitely that deserves better impl. I consider that temporary solution

File: adapters/repos/db/index.go (hunk 2 above)
Comment (aliszka) [reply adapters/repos/db/index.go:323]: Don't like it either. Also to be refactored.

**Pull request merged**